core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -693,9 +693,10 @@ public String toString(@Nullable AbstractBlockChain chain) {
             }
             s.append(String.format(Locale.US, "%n"));
         }
-        Coin fee = getFee();
+        final Coin fee = getFee();
         if (fee != null)
-            s.append("     fee  ").append(fee.toFriendlyString()).append(String.format(Locale.US, "%n"));
+            s.append("     fee  ").append(fee.multiply(1000).divide(unsafeBitcoinSerialize().length).toFriendlyString())
```

Comment:
Thanks for your review! First of all, I believe fee is always calculated per 1000 bytes, rather than 1024 bytes. Please correct me if I'm wrong.

The Coin class internally stores satoshis, but `.toFriendlyString()` prints BTC values currently. So in the end you'll see values like '0.00044 BTC/kB'. That's consistent with block explorers, e.g. https://blockexplorer.com/tx/05e1b533e63b8c93d0944e3b024a52a22677664e9c2b63364e336208d56db093

Lastly, I think you meant `/1024` rather than `*1024`. I didn't run your code but I believe it will output much too small values. Also, if you do it like that you'll run into rounding problems so I prefer to multiply first, then divide. Any thoughts?